jena-base/src/main/java/org/apache/jena/atlas/io/IO.java

```diff
@@ -86,13 +86,34 @@ static public InputStream openFileEx(String filename) throws IOException, FileNo
             filename = filename.substring("file:".length());
             filename = IRILib.decodeHex(filename);
         }
-        InputStream in = new FileInputStream(filename);
+        InputStream in0 = new FileInputStream(filename);
+        InputStream in = IO.ensureBuffered(in0);
         String ext = getExtension(filename);
+
+        // Input is a file stream.
+        // https://commons.apache.org/proper/commons-compress/examples.html#Buffering :
+        // """
+        // The stream classes all wrap around streams provided by the calling
+        // code and they work on them directly without any additional
+        // buffering. On the other hand most of them will benefit from
+        // buffering so it is highly recommended that users wrap their stream
+        // in Buffered(In|Out)putStreams before using the Commons Compress
+        // API.
+        // """
+        // GZip and Snappy have internal buffering.
+        // BZip2 does not.
         switch ( ext ) {
-            case "":        return in;
-            case ext_gz:    return new GZIPInputStream(in);
-            case ext_bz2:   return new BZip2CompressorInputStream(in, true);
-            case ext_sz:    return new SnappyCompressorInputStream(in);
+            case "":
+                return in;
+            case ext_gz:
+                // Makes a small improvement (<5%) to use 8K.
+                return new GZIPInputStream(in, 8*1024);
+            case ext_bz2:
+                // Make a huge improvement. x10 faster.
+                in = IO.ensureBuffered(in);
```

Comment:
@afs this part is looking weird.

We have earlier called `InputStream in = IO.ensureBuffered(in0);`. Then if bz2, we call `in = IO.ensureBuffered(in);` again? Like twice calling `IO.ensureBuffered`?

Reply:
`ensureBuffered()` checks whether the passed input stream is already buffered and if so is a no-op so don't think it matters if we call it twice